Snowflake.Data.Tests/SFConnectionPoolT.cs

```diff
@@ -504,6 +504,98 @@ public void TestConcurrentConnectionPoolingAsync()
             ConcurrentPoolingAsyncHelper(connStr, true);
         }
 
+        [Test]
+        public void TestRollbackTransactionOnPooledWhenExceptionOccurred()
+        {
+            SnowflakeDbConnectionPool.SetPooling(true);
+            SnowflakeDbConnectionPool.SetMaxPoolSize(10);
+
+            object firstOpenedSession = null;
+            using (var connection = new SnowflakeDbConnection())
+            {
+                connection.ConnectionString = ConnectionString;
+                connection.Open();
+                firstOpenedSession = connection.SfSession.sessionId;
+                connection.BeginTransaction();
+                Assert.Throws<SnowflakeDbException>(() =>
+                {
+                    using (var command = connection.CreateCommand())
+                    {
+                        command.CommandText = "invalid command will throw";
+                        command.ExecuteNonQuery();
+                    }
+                });
+            }
+
+            using (var connectionWithSessionReused = new SnowflakeDbConnection())
+            {
+                connectionWithSessionReused.ConnectionString = ConnectionString;
+                connectionWithSessionReused.Open();
+                Assert.AreEqual(firstOpenedSession, connectionWithSessionReused.SfSession.sessionId);
```

Comment:
pool has 10 elements - are we sure that it will be always the same connection?

Reply:
I will set pool size to one as @sfc-gh-pfus  gave good suggestion on this. This way even if we change future logic of the pool reusing a single connection should be covered in this test.